util/gitutil/git_ref.go

```diff
@@ -67,10 +66,7 @@ func ParseGitRef(ref string) (*GitRef, error) {
 			Path:   strings.TrimPrefix(ref, "github.com/"),
 		})
 	} else {
-		remote, err = ParseURL(ref)
-		if errors.Is(err, ErrUnknownProtocol) {
-			return nil, err
-		}
+		remote, err = ParseURL(ref, NoImplicitSCPUsername())
```

Comment:
I think we should protect this with a LLB capability.

Reply:
Added a cap for this, we add the new `pb.CapSourceGitImplicitUsername` capability if we get an scp-style url that has no username associated. This means we can't have a new frontend send a username without a username that an old server won't understand.